### PR TITLE
ISSUE-1644 Check if the shorten belong to the user before update / delete

### PR DESCRIPTION
**Pull request by jecos**

### CODE REVIEW

File: src/main/java/bio/overture/riff/service/RiffService.java [hunk 1]
```diff
@@ -75,7 +75,7 @@ public RiffResponse makeRiff(JWTUser user, ShortenRequest request) {
   }
 
   public boolean deleteRiff(JWTUser user, String id) {
-    val optionalRiff = repository.findById(Long.valueOf(id, 36));
+    val optionalRiff = repository.findByUidAndId(user.getUid(), Long.valueOf(id, 36));
     if (optionalRiff.isPresent()) {
       val riff = optionalRiff.get();
       if (user.getUid().equals(riff.getUid())) {
```
Comment (joneubank) [inline src/main/java/bio/overture/riff/service/RiffService.java:81]: Looks like this line provides the logic to confirm that the riff entry matches the user id from the request. If so, we don't need any modification tot he repository fetch request.

Perhaps all we needed to do was add a check like this to the updateRiff method.

**Review by joneubank — CHANGES_REQUESTED** [review]
Find by Uid and Id doesn't accomplish anything except filtering by userId at the fetch layer, which should actually be maintained in our auth rules or application logic.

The deletRiff method already has a check to deal with this, and we should consider adding that to updateRiff instead of putting this filter in the repository fetch.

File: src/main/java/bio/overture/riff/service/RiffService.java (hunk 1 above)
Comment (andricDu) [inline src/main/java/bio/overture/riff/service/RiffService.java:81]: If we are retrieving this riff with the uid as part of the query, do we still need this conditional here?

**Review by andricDu — CHANGES_REQUESTED** [review]

File: src/main/java/bio/overture/riff/service/RiffService.java (hunk 1 above)
Comment (jecos) [reply src/main/java/bio/overture/riff/service/RiffService.java:81]: I missed that, I'll change the code to have the same thing than the delete

**Comment by jecos** [comment]
I closed this PR to fork from kids-first repo, not Ste-Justine one

**Pull request closed without merging**